### PR TITLE
Cf configurable

### PR DESCRIPTION
**Pull request by Sp1l**
In environments where there's issues accessing 1.1.1.1 or IP-addresses, it is helpful to be able to override the Cloudflare lookup and purge endpoints.

I've just been testing the endpoints a bit, and it seems that https://cloudflare-dns.com exposes the same purge API as https://1.0.0.1 and https://1.1.1.1 do. Perhaps it's best to just use that in stead of IP-addresses!

### CODE REVIEW

File: acme.sh [hunk 1]
```diff
@@ -3613,7 +3617,11 @@ _ns_purge_cf() {
   _cf_d="$1"
   _cf_d_type="$2"
   _debug "Cloudflare purge $_cf_d_type record for domain $_cf_d"
-  _cf_purl="https://1.1.1.1/api/v1/purge?domain=$_cf_d&type=$_cf_d_type"
+  if [ "$CLOUDFLARE_DOHJSON_URL" != "" ]; then
+    _cf_purl="$ONE_ONE_ONE_ONE_URL?domain=$_cf_d&type=$_cf_d_type"
```
Comment (Neilpang) [inline acme.sh]: what is the value of ONE_ONE_ONE_ONE_URL ?

Comment (Sp1l) [reply acme.sh]: Replaced it with CLOUDFLARE_DOHJSON_URL to make it more explicit.

For networks where 1.1.1.1 is unusable, set it to `1001.cloudflare-dns.com` (see #2330)

**Comment by Sp1l** [comment]
> I've just been testing the endpoints a bit, and it seems that https://cloudflare-dns.com exposes the same purge API as https://1.0.0.1 and https://1.1.1.1 do. Perhaps it's best to just use that in stead of IP-addresses!

Response is the same as on one.one.one.one:
```
$ host cloudflare-dns.com
cloudflare-dns.com has address 104.16.249.249
cloudflare-dns.com has address 104.16.248.249
cloudflare-dns.com has IPv6 address 2606:4700::6810:f9f9
cloudflare-dns.com has IPv6 address 2606:4700::6810:f8f9
$
$ curl -vd "" "https://cloudflare-dns.com/api/v1/purge?domain=acme.sh&type=A"
*   Trying 104.16.249.249:443...
* TCP_NODELAY set
* Connected to cloudflare-dns.com (104.16.249.249) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/pki/tls/certs/ca-bundle.crt
  CApath: none
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384
* ALPN, server accepted to use h2
* Server certificate:
*  subject: C=US; ST=California; L=San Francisco; O=Cloudflare, Inc.; CN=cloudflare-dns.com
*  start date: Jan 28 00:00:00 2019 GMT
*  expire date: Feb  1 12:00:00 2021 GMT
*  subjectAltName: host "cloudflare-dns.com" matched cert's "cloudflare-dns.com"
*  issuer: C=US; O=DigiCert Inc; CN=DigiCert ECC Secure Server CA
*  SSL certificate verify ok.
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* Using Stream ID: 1 (easy handle 0x5629c5f04700)
> POST /api/v1/purge?domain=acme.sh&type=A HTTP/2
> Host: cloudflare-dns.com
> User-Agent: curl/7.65.3
> Accept: */*
> Content-Length: 0
> Content-Type: application/x-www-form-urlencoded
> 
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* old SSL session ID is stale, removing
* Connection state changed (MAX_CONCURRENT_STREAMS == 256)!
< HTTP/2 200 
< date: Mon, 29 Jul 2019 11:21:28 GMT
< content-type: text/plain; charset=utf-8
< content-length: 87
< x-content-type-options: nosniff
< x-envoy-upstream-service-time: 0
< strict-transport-security: max-age=31536000; includeSubDomains; preload
< expect-ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
< server: cloudflare
< cf-ray: 4fdeb5a26f85732f-AMS
< 
Purge request queued. Please wait a few seconds and verify the request was successful.
* Connection #0 to host cloudflare-dns.com left intact
```

**Comment by Sp1l** [comment]
Looks like this is all the exact same service. Looking at the Subject-Alt-Names:
```
CN=cloudflare-dns.com
DNS Name: cloudflare-dns.com
DNS Name: *.cloudflare-dns.com
DNS Name: one.one.one.one
IP Address: 1.1.1.1
IP Address: 1.0.0.1
IP Address: 162.159.132.53
IP Address: 2606:4700:4700::1111
IP Address: 2606:4700:4700::1001
IP Address: 2606:4700:4700::64
IP Address: 2606:4700:4700::6400
IP Address: 162.159.36.1
IP Address: 162.159.46.1
```

**Comment by Neilpang** [comment]
Thank you,  I just made a fix.  please have a try with the dev branch.

Thanks.

**Pull request closed without merging**